### ISSUE TITLE
Duplicated rendering the page when the URL contains query string

### ISSUE THREAD

**Issue by hyurl** [vercel/next.js#20727]
# Bug report

## Describe the bug

When the URL contains the query string, the page will be rendered twice when the page is refreshed by clicking the Reload button.

## To Reproduce

For Example, the following page component:

```ts
// pages/example.tsx
export default function Example() {
    console.log("foo bar");
    return <div>Hello, World!</div>
}
```

1. Go to 'http://localhost:4000/example', the page renders fine and the console logs only once of the `foo bar`;
2. Then we add a query string, for instance, 'http://localhost:4000/example?foo=bar' and hit Goto, this time the page seems to render fine but the console actually logs twice of the `foo bar`, meaning the page has rendered twice/duplicated.

## Expected behavior

The page should only render once regardless of having the query string or not.

This bug also makes `react-redux` not working properly, when duplicated rendering, react-redux returns the state twice before it actually updated, and the dispatch function will not work after the second return, meaning if we want to use the dispatch function to update the state hosted by react-redux inside a `useEffect` hook, it won't work.

## Screenshots

None

## System information

- OS: Windows
- Browser: Chrome
- Version of Next.js: 10.0.4/10.0.3
- Version of Node.js: 14
- Deployment: next start

## Additional context

None

**Comment by ijjk** [comment]
Hi, this is the expected behavior and not a bug as described in the documentation here https://nextjs.org/docs/advanced-features/automatic-static-optimization#how-it-works

**Comment by hyurl** [comment]
@ijjk But I do not have either `getServerSideProps` or `getInitialProps` in the page, there is a `getInitialProps` in the `_document.tsx` though, and how do we solve the conflict with react-redux.

**Comment by Timer** [comment]
You can ask for help in our [Discord](https://nextjs.org/discord)!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
